modules/commons/src/main/resources/schemas/neurosciencegraph/commons/annotation/v0.2.0.json

```diff
@@ -0,0 +1,202 @@
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "SelectorTargetShape"
+        },
+        {
+          "property": [
+            {
+              "path": "oa:hasSelector",
+              "node": "{{base}}/schemas/neurosciencegraph/commons/selectors/v0.2.0/shapes/TextPositionSelectorShape",
+              "minCount": 1,
+              "maxCount": 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id": "FigureTargetShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "SelectorTargetShape"
+        },
+        {
+          "property": [
+            {
+              "path": "oa:hasSelector",
+              "node": "{{base}}/schemas/neurosciencegraph/commons/selectors/v0.2.0/shapes/FigureSelectorShape",
+              "minCount": 1,
+              "maxCount": 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id": "EquationTargetShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "SelectorTargetShape"
+        },
+        {
+          "property": [
+            {
+              "path": "oa:hasSelector",
+              "node": "{{base}}/schemas/neurosciencegraph/commons/selectors/v0.2.0/shapes/EquationSelectorShape",
+              "minCount": 1,
+              "maxCount": 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id": "TableTargetShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "SelectorTargetShape"
+        },
+        {
+          "property": [
+            {
+              "path": "oa:hasSelector",
+              "node": "{{base}}/schemas/neurosciencegraph/commons/selectors/v0.2.0/shapes/TableSelectorShape",
+              "minCount": 1,
+              "maxCount": 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id": "AreaTargetShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "SelectorTargetShape"
+        },
+        {
+          "property": [
+            {
+              "path": "oa:hasSelector",
+              "node": "{{base}}/schemas/neurosciencegraph/commons/selectors/v0.2.0/shapes/AreaSelectorShape",
+              "minCount": 1,
+              "maxCount": 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id": "BodyShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI"
+    },
+    {
+      "@id": "ContributionShape",
+      "@type": "sh:NodeShape",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "property": [
+        {
+          "path": "prov:agent",
+          "name": "Agent",
+          "nodeKind": "sh:IRI",
+          "class": "prov:Agent",
+          "minCount": 1
+        }
+      ]
+    }
```

Comment:
Instead of creating the ContributionShape here, the existing agent shape https://github.com/INCF/neuroshapes/blob/master/modules/core/src/main/resources/schemas/neurosciencegraph/core/agent/v0.1.0.json could be used as value of the node "nsg:contribution" in the property shapes of this schema

Reply:
I did it to enforce the fact that we want to have IRI of agents so that they come from a defined list, i.e. are not defined in this payload. Besides, the existing shapes for agents are not in a good 'shape':
https://github.com/BlueBrain/nexus-prov/blob/master/modules/prov/src/main/resources/schemas/nexus/provsh/agent/v1.0.0.json
https://github.com/BlueBrain/nexus-prov/blob/master/modules/prov/src/main/resources/schemas/nexus/prov/agent/v1.0.0.json